credentials/credentials.go

```diff
@@ -50,6 +50,23 @@ type PerRPCCredentials interface {
 	RequireTransportSecurity() bool
 }
 
+// SecurityLevel defines the protection level on an established channel.
+type SecurityLevel int
+
+const (
+	// SecurityNone indicates a channel is a plaintext channel that does not provide any protection.
+	SecurityNone SecurityLevel = iota
+	// IntegrityOnly indicates a channel only provides integrity protection.
+	IntegrityOnly
+	// PrivacyAndIntegrity indicates a channel provides both privacy and integrity protection.
+	PrivacyAndIntegrity
+)
+
+// CommonAuthInfo contains authenticated information common to AuthInfo implementations.
```

Comment:
Here's an idea:

```go
package credentials

type CommonAuthInfo struct {
   SecurityLevel SecurityLevel
}

func (c CommonAuthInfo) CommonInfo() CommonAuthInfo {
    // TODO: Use pointer receiver/return value??
	return c
}

package mycreds

type MyAuthInfo struct {
   // my stuff
   CommonAuthInfo
}

ai := &MyAuthInfo{CommonAuthInfo: CommonAuthInfo{SecurityLevel: 5}}
```

Now `MyAuthInfo` automatically implements the necessary interface.  So all users have to do if they want to use it is embed it in their existing `AuthInfo` implementation.

I haven't fully thought this through but this seems promising.  WDYT?  CC @menghanl

Reply:
The proposal (i.e., using composition) is interesting. I do not foresee any `AuthInfo` implementations will need different realizations for `CommonInfo()` which will be just an accessor. I will go ahead update the PR unless @menghanl thinks differently.